stdlib/map.mc

```diff
@@ -14,9 +14,8 @@ let mapLookupOpt = lam eq. lam k. lam seq. findAssoc (eq k) seq
 -- Value binded to k in seq, if no such binding in seq then an error occurs.
 -- Equality defined by eq.
 let mapLookup = lam eq. lam k. lam seq.
-  optionMapOrElse (mapLookupOpt eq k seq)
-                  (lam _. error "Element not found")
-                  (lam x. x)
+  optionGetOrElse (lam _. error "Element not found")
```

Comment:
The changes looks good to me! (Also verified that the tests succeed after a merge with develop.)

More of a philosophical discussion: Could it be an idea to adopt a standard of how to handle error prints? At the moment (and previously) the error just prints out that something went wrong, but no details on what actually went wrong. In this case we could for example print out which key that was missing from the map:
```ocaml
let mapLookup = lam eq. lam k. lam seq.
  optionGetOrElse (lam _. error (strJoin " " ["Element bound to key \"", k, "\" was not found"]))
                  (mapLookupOpt eq k seq)
```

Or could we perhaps adopt a formatted error print? Such as:
```ocaml
let mapLookup = lam eq. lam k. lam seq.
  optionGetOrElse (lam _. errorFmt "Element bound to key \"$0\" was not found" [k])
                  (mapLookupOpt eq k seq)
```